### DISCUSSION TITLE
Test-DbaLastBackup read .BAK from external location

### DISCUSSION THREAD

**Discussion by christophorosp** [dataplat/dbatools#7739] [Ideas] [answered]
Is there a way to have Test-DbaLastBackup read the .BAK files from an external location such as an external backup storage repo?

We have backups being copied from the local backup disk to a central backup location (stores bak files from multiple servers) and would love to have Test-DbaLastBackup batch-restore and test backups to a LAB server

i.e. supply a -Path parameter pointing out Test-DbaLastBackup to retrieve the backup file of the instance, instead of the default backup location as is right now

**Comment by niphlod** [comment, marked as answer]
once you move files around, there's no reason to look at the history to restore the LAST copy, and that's what Test-DbaLastBackup does. 
It's not difficult to test restores without Test-DbaLastBackup (in fact, you can read what it does pretty easily):
- do a Restore-DbaDatabase (RestoredDatabaseNamePrefix, DestinationFilePrefix, VerifyOnly, EnableException are your best friends here )
- if you do a full restore and if you need a checkdb, issue a Start-DbccCheck
- if you restored the db and everything goes right, Remove-DbaDatabase

**Reply by christophorosp** [reply]
Thanks for the reply, this is quite useful.

> if you do a full restore and if you need a checkdb, issue a Start-DbccCheck

How can this be done? This is an internal function, no? 
Also will i be able to save using ConvertTo-DbaDataTable and write to database table using the Start-DbccCheck as with Test-DbaLastBackup?

**Reply by niphlod** [reply]
whoops, you're right. But Start-DbccCheck just issues a query `$server.Query("DBCC CHECKDB ([$DbName])")` . As for saving results, you can "mock" an object with the same properties of Test-DbaLastBackup and surely it'll be "accepted" by ConvertTo-DbaDataTable.
